Assets/MixedRealityToolkit/Interfaces/Registrars/IMixedRealityServiceRegistrar.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Core.Definitions.Utilities;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.MixedReality.Toolkit.Core.Interfaces
+{
+    /// <summary>
+    /// Interface for Mixed Reality Toolkit service registration.
+    /// </summary>
+    public interface IMixedRealityServiceRegistrar
+    {
+        #region IMixedRealityService registration
+
+        /// <summary>
+        /// Registers a service of the specified type.
+        /// </summary>
+        /// <typeparam name="T">The interface type of the service to be registered (ex: IMixedRealityBoundarySystem).</typeparam>
+        /// <param name="serviceInstance">An instance of the service to be registered.</param>
+        bool RegisterService<T>(T serviceInstance) where T : IMixedRealityService;
+
+        /// <summary>
+        /// Registers a service of the specified type.
+        /// </summary>
+        /// <typeparam name="T">The interface type of the service to be registered (ex: IMixedRealityBoundarySystem).</typeparam>
+        /// <param name="concreteType">The concrete type to instantiate.</param>
+        /// <param name="supportedPlatforms">The platform(s) on which the service is supported.</param>
```

Comment:
Is it safe to assume that for the function above, SupportedPlatforms will pretty much be ALL (i.e. even if I add a new enum type in the future to SupportedPlatforms, does that mean that people who call RegisterService with the function above will implicitly get pulled into supporting those new enum types?)

Reply:
Woops sent too soon - by function above, I meant the first overload that just takes a type :)

Reply:
its probably safer to require explicit platform declarations compared to having a global all. But its kind of tedious making these sorts of updates.

I would imagine that either choice could cause issues:
1) Without global all - Someone starts using a new platform, a change is required to all services to start supporting said platform 
2) With global all - Someone starts using a new platform, first use of specific services results in needing to remove platform support for said service. This removes global all/all additional platforms again start requiring an update to this specific service

It kind of seems like 1 is safer even if its more leg work. 2 seems like it would generate more frustration for developers/may generate stranger behaviors when out in the wild.

Reply:
this is the current service locator behavior. it was a primary goal to minimize functional change when applying this PR.

the team is definitely willing to entertain thoughts of how to make improvements to the existing locator, feel free to file issues and they can be addressed in the not too distant future